salt/apps/ocw/templates/webhook-publish.sh.jinja

```diff
@@ -45,24 +60,37 @@ else
     rm $retry_file
 fi
 
+cd /opt/ocw/hugo-course-publisher \
+    || error_and_exit "Can not cd to hugo-course-publisher!"
+
+orig_commit=`git rev-parse HEAD`
+log_message "Pulling hugo-course-publisher"
+git pull || error_and_exit "Can not pull hugo-course-publisher"
+new_commit=`git rev-parse HEAD`
+log_message "$orig_commit -> $new_commit"
+
+if [ $option == "full" ]; then
+    if [ "$orig_commit" != "$new_commit" ]; then
+        log_message "Pulled new commit, so continuing with publish."
+    else
+        error_and_exit "No new commit or got a rev-parse error, so stopping here."
```

Comment:
🤔 On second thought, maybe this should not exit with an error code.

Reply:
I agree that it should just `exit 0` in the event of an unchanged commit.

Reply:
Changed in commit a6dd1c76